x-pack/test/accessibility/apps/search_profiler.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export default function({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common', 'security']);
+  const testSubjects = getService('testSubjects');
+  const aceEditor = getService('aceEditor');
+  const a11y = getService('a11y');
+  const flyout = getService('flyout');
+
+  describe('Accessibility Search Profiler Editor', () => {
+    before(async () => {
+      await PageObjects.common.navigateToApp('searchProfiler');
+      await a11y.testAppSnapshot();
+      expect(await testSubjects.exists('searchProfilerEditor')).to.be(true);
+    });
+
+    it('input the JSON in the aceeditor', async () => {
+      const input = {
+        query: {
+          bool: {
+            should: [
+              {
+                match: {
+                  name: 'fred',
+                },
+              },
+              {
+                terms: {
+                  name: ['sue', 'sally'],
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          stats: {
+            stats: {
+              field: 'price',
+            },
+          },
+        },
+      };
+
+      await aceEditor.setValue('searchProfilerEditor', JSON.stringify(input));
+      await a11y.testAppSnapshot();
+    });
+
+    it('click on the profile button', async () => {
+      await testSubjects.click('profileButton');
+      await a11y.testAppSnapshot();
+    });
+
+    it('click on the view details link', async () => {
+      const viewShardDetailslink = await testSubjects.findAll('viewShardDetails');
+      await viewShardDetailslink[0].click();
+      await a11y.testAppSnapshot();
+    });
+
+    it('close the fly out', async () => {
+      await flyout.ensureAllClosed();
+      await a11y.testAppSnapshot();
+    });
+
```

Comment:
@rasroh pulled it down in my local and just tested it. Awesome tests. One tiny test is missing though. Can you please add a step here for closing the "View details" after you close the flyout? In particular closing of this?  That is one user action possible. Thanks!
<img width="1792" alt="close_Details" src="https://user-images.githubusercontent.com/7074629/78990709-23914f80-7b05-11ea-932b-ca3425957860.png">

Reply:
added the test.